src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,147 @@
+2. Lack of Comprehensive Support Documentation: Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
+## Keycloak Installation
+### By Docker image
+Before we start, lets make sure following things get installed into your system.
+- install [Docker Engine](https://docs.docker.com/engine/install/)
+- install [Docker Compose](https://docs.docker.com/compose/install/)
+
+1. Keycloak using docker image
+Run the following docker command to serve the Keycloak image. Keycloak will be available on `http://localhost:8080.
+```bash
+docker run -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:24.0.2 start-dev
+ ```
+
+2. Keycloak with postgres database using docker-compose
+```
+services:
+  postgres:
+    image: postgres:alpine
+    volumes:
+      - postgres_data:/var/lib/postgresql/data
+    environment:
+      POSTGRES_DB: keycloak
+      POSTGRES_USER: keycloak
+      POSTGRES_PASSWORD: keycloak
+
+  keycloak:
+    image: quay.io/keycloak/keycloak:24.0.1
+    command: ["start-dev"]
+    environment:
+      KC_DB: postgres
+      KC_DB_URL: "jdbc:postgresql://postgres:5432/keycloak"
+      KC_DB_USERNAME: keycloak
+      KC_DB_PASSWORD: keycloak
+      KC_FEATURES: impersonation
+      KEYCLOAK_ADMIN: admin
+      KEYCLOAK_ADMIN_PASSWORD: admin
+    ports:
+      - 8001:8080 # map keycloak port 8080 to 8001
+    depends_on:
+      - postgres
+
+volumes:
+  postgres_data:
+```
+
+Save above code in `compose.yaml` file and run `docker compose up` in terminal.
+
+Docker compose serves Keycloak on `http://localhost:8001`
+
+### By distribution file
+There is another to set up keycloak using keycloak distribution file.
+1. Download the keycloak distribution
+    ```
+    export KC_VERSION=24.0.4
+    curl -LO  https://github.com/keycloak/keycloak/releases/download/"${KC_VERSION}"/keycloak-"${KC_VERSION}".zip
+    ```
+
+2. Unzip the keycloak distribution package
+    ```
+    unzip keycloak-${KC_VERSION}.zip
+    ```
+
+3. Install openjdk
+    ```
+    sudo apt install openjdk-17-jdk
+    ```
+
+4. Navigate to Keycloak directory
+    ```
+    cd keycloak-${KC_VERSION}
+   ```
+
+5. When we start the server for the first time, we have to set the admin user and the admin password:
+    ```
+   KEYCLOAK_ADMIN=admin KEYCLOAK_ADMIN_PASSWORD=admin ./bin/kc.sh start-dev
+   ```
+    > Note: if port 8080 is already used then you can map keycloak port using Keycloak environment variable 'KC_HTTP_PORT=8001'
+
+6. When we start again, it is not necessary to set these variables, again. You can start the server with:
+    ```
+   ./bin/kc.sh start-dev
+   ```
+   > Note: start-dev runs the quarkus application in DEV-mode. Do not use this for Production.
```

Comment:
What is "quarkus"?
When reading this note, I have no idea what it has to do with Keycloak.